### PR TITLE
refactor "isGeneratorFunction"

### PR DESCRIPTION
**Pull request by zensh**
fix for object that don't have `constructor` and `GeneratorFunction` that it's displayName is not "GeneratorFunction"

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -213,12 +213,9 @@ function isGenerator(obj) {
  */
 function isGeneratorFunction(obj) {
   var constructor = obj.constructor;
-  var proto = constructor.prototype;
-  var name = constructor.displayName || constructor.name;
-  var nameLooksRight = 'GeneratorFunction' == name;
-  var methodsLooksRight = 'function' == typeof proto.next &&
-                          'function' == typeof proto.throw;
-  return nameLooksRight || methodsLooksRight;
+  if (!constructor) return false;
```
Comment (madbence) [inline index.js]: Is there a case when `obj` doesn't have `constructor`?

**Comment by madbence** [comment]
I really like that it uses `isGenerator` :+1:

File: index.js (hunk 1 above)
Comment (zensh) [inline index.js]: Object.create(null), use it as caching object

Comment (madbence) [inline index.js]: :+1: indeed

**Comment by jonathanong** [comment]
lol damn. didn't think of this

**Comment by jonathanong** [comment]
1e50cefc85de4570d394e3f9b0a32a7a5db7e924

**Pull request closed without merging**